### PR TITLE
latency test enhancement - part 1

### PR DESCRIPTION
**Pull request by min-xu-et**
This is part 1 of change to make latency test generate graph.

* this makes the batch_size arg a list, but user can continue use it the same way as before when specifying a single value
* this optionally dumps the results in a json file. example:

```
$ cat out.json
[
  {
    "batch_size": 1,
    "output_len": 256,
    "prefill_latency": 0.12703561782836914,
    "prefill_throughput": 4030.3657253962833,
    "avg_decode_latency": 0.05401019565761089,
    "avg_decode_throughput": 18.515022725326567,
    "total_latency": 13.953645706176758,
    "total_throughput": 55.03937939745991
  },
  {
    "batch_size": 1,
    "output_len": 256,
    "prefill_latency": 0.137282133102417,
    "prefill_throughput": 3729.545778677777,
    "avg_decode_latency": 0.054004695266485214,
    "avg_decode_throughput": 18.516908484818174,
    "total_latency": 13.962484121322632,
    "total_throughput": 55.00453882895798
  }
]
```

### CODE REVIEW

File: python/sglang/bench_latency.py [hunk 1]
```diff
@@ -218,15 +234,25 @@ def latency_test(
         f"max_batch_size={model_runner.max_total_num_tokens // (bench_args.input_len + bench_args.output_len)}"
     )
 
+    # To make this PR easier to review, for now, only do the first element in batch_size tuple.
+    bench_args.batch_size = bench_args.batch_size[0]
```
Comment (merrymercy) [inline python/sglang/bench_latency.py:237]: Remove this line or do you want to do it in the follow up PR?

File: python/sglang/bench_latency.py [hunk 2]
```diff
@@ -267,13 +297,25 @@ def run_once(output_len):
...
+    ret = []
+    ret.append(run_once(bench_args.output_len))
+
+    # Write results
+    if bench_args.result_filename:
+        data = []
+        if os.path.isfile(bench_args.result_filename):
+            data = json.loads(open(bench_args.result_filename, "r").read())
+        data += ret
+        open(bench_args.result_filename, "w").write(json.dumps(data, indent=2))
```
Comment (merrymercy) [inline python/sglang/bench_latency.py]: Can we use jsonl format with append mode?

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/bench_latency.py (hunk 1 above)
Comment (min-xu-et) [reply python/sglang/bench_latency.py:237]: That's my thought to make the review easier. If you prefer, I can include in this PR another commit for the part 2 as well. For part 2, I plan to move the nested functions out of this test function and call them in a loop.

File: python/sglang/bench_latency.py (hunk 2 above)
Comment (min-xu-et) [reply python/sglang/bench_latency.py]: Great idea. Will do.

Comment (min-xu-et) [reply python/sglang/bench_latency.py]: resolved in the new commit https://github.com/sgl-project/sglang/pull/909/commits/f52110d52b4b96ced2dbd25dadc8c90d99af5ad6

File: python/sglang/bench_latency.py (hunk 1 above)
Comment (min-xu-et) [reply python/sglang/bench_latency.py:237]: please unresolve this if you like to combine step 1 and step 2 in this PR.

**Review by Ying1123 — APPROVED** [review]

**Pull request merged**